subworkflows/nf-core/gatk4/joint_germline_variant_calling/main.nf

```diff
@@ -1,92 +1,140 @@
-    annotation                // channel: [annotations, to, use, for, variantrecal, filtering]
-    mode                      // channel: which mode to run variantrecal: SNP/INDEL/BOTH
-    create_rscript            // channel: true/false whether to generate rscript plots in variantrecal
-    truthsensitivity          // channel: 0-100.0 truthsensitivity cutoff for applyvqsr
+    input            // channel: [ val(meta), [ input ], [ input_index ], interval, [], []]
+    fasta            // channel: /path/to/reference/fasta
+    fai              // channel: /path/to/reference/fasta/index
+    dict             // channel: /path/to/reference/fasta/dictionary
+    dbsnp
+    dbsnp_tbi
+    dbsnp_vqsr
+    resource_indels_vcf
+    resource_indels_tbi
+    known_indels_vqsr
+    resource_snps_vcf
+    resource_snps_tbi
+    known_snps_vqsr
 
     main:
-    ch_versions       = Channel.empty()
+    ch_versions    = Channel.empty()
+
+    //
+    //Map input for GenomicsDBImport.
+    //rename based on num_intervals, group all samples by their interval_name/interval_file and restructure for channel
+    //
+    gendb_input = input.map{
+        meta, gvcf, tbi, intervals->
+            new_meta = [id: "joint_variant_calling",
+                        intervals_name: meta.intervals_name,
+                        num_intervals: meta.num_intervals
+                       ]
+            [ new_meta, gvcf, tbi, intervals ]
+        }.groupTuple(by:[0,3]).map{ new_meta, gvcf, tbi, intervals ->
+            interval_file = new_meta.num_intervals > 1 ? intervals : params.intervals
+            [ new_meta, gvcf, tbi, interval_file, [], [] ] }
 
     //
     //Convert all sample vcfs into a genomicsdb workspace using genomicsdbimport.
     //
-    // gendb_input       = .combine([interval_file]).combine(['']).combine([dict])
+    GATK4_GENOMICSDBIMPORT ( gendb_input, false, false, false )
 
-    // GENOMICSDBIMPORT ( gendb_input, false, false, false )
-    // ch_versions       = ch_versions.mix(GENOMICSDBIMPORT.out.versions)
+    genotype_input = GATK4_GENOMICSDBIMPORT.out.genomicsdb.map{
+        meta, genomicsdb ->
+            [meta, genomicsdb, [], [], []]
+        }
+    ch_versions = ch_versions.mix(GATK4_GENOMICSDBIMPORT.out.versions)
 
     //
     //Joint genotyping performed using GenotypeGVCFs
+    //Sort vcfs called by interval within each VCF
     //
-    // ch_genotype_in    = GENOMICSDBIMPORT.out.genomicsdb.collect()
-    // //[] is a placeholder for the input where the vcf tbi file would be passed in for non-genomicsdb workspace runs, which is not necessary for this workflow as it uses genomicsdb workspaces.
-    // ch_genotype_in.add([])
 
-    // GENOTYPEGVCFS ( ch_genotype_in, fasta, fai, dict, sites, sites_index )
-    // ch_versions       = ch_versions.mix(GENOTYPEGVCFS.out.versions)
+    vcfs = GATK4_GENOTYPEGVCFS ( genotype_input, fasta, fai, dict, dbsnp, dbsnp_tbi).vcf
```

Comment:
```suggestion
    vcfs = GATK4_GENOTYPEGVCFS ( genotype_input, fasta, fai, dict, dbsnp, dbsnp_tbi).out.vcf
```

I know the other hitng also works, but for consistency adding `out` would be nice

Reply:
Ran into the same thing, and this seems to be the way to go